firebase/FirestoreCtrl.ts

```diff
@@ -320,6 +327,151 @@ export default class FirestoreCtrl {
     }
   }
 
+  /**
+   * Retrieves all users from Firestore.
+   * @returns A promise that resolves to an array of users.
+   * */
+
+  async getAllUsers(): Promise<DBUser[]> {
+    try {
+      const usersRef = collection(firestore, "users");
+      const querySnapshot = await getDocs(usersRef);
+      const users = querySnapshot.docs.map((doc) => {
+        const data = doc.data();
+        return {
+          ...data,
+        } as DBUser;
```

Comment:
Why is the return a DBUsers? This is very computationally expensive. It would take a significant amount of time to run for every search.
Instead use the query method in firebase to obtain the user_name instead, you can also search of contained the regex written as the searched text. This way you don't have to retrieve all users every time you query.

https://firebase.google.com/docs/firestore/query-data/queries

Reply:
Because I actually need more informations than just the username, I also need the profile picture and it allows me to easily change informations in case of add/unadd etc.

Reply:
Also this is what we return in every get methods we did in FirestoreCtrl. Sure it is expensive, but even if I only retrieve the username I'll have to getUser for every username anyway which will be the same.